Tests/NumLib/TestExtrapolation.cpp

```diff
@@ -0,0 +1,359 @@
+    getShapeMatrix(const unsigned integration_point) const override
+    {
+        auto const& N = _shape_matrices[integration_point].N;
+
+        // assumes N is stored contiguously in memory
+        return Eigen::Map<const Eigen::VectorXd>(N.data(), N.size());
+    }
+
+    std::vector<double> const&
+    getIntegrationPointValues(IntegrationPointValue const property,
+                              std::vector<double>& cache) const override
+    {
+        switch (property)
+        {
+        case IntegrationPointValue::SavedQuantity:
+            return _int_pt_values;
+        case IntegrationPointValue::DerivedQuantity:
+            cache.clear();
+            for (auto value : _int_pt_values)
+                cache.push_back(2.0 * value);
+            return cache;
+        }
+
+        std::abort();
+    }
+
+    void interpolateNodalValuesToIntegrationPoints(
+            std::vector<double> const& local_nodal_values,
+            IntegrationPointValue const property) override
+    {
+        switch (property)
+        {
+        case IntegrationPointValue::SavedQuantity:
+            ::interpolateNodalValuesToIntegrationPoints(
+                        local_nodal_values, _shape_matrices, _int_pt_values);
+            break;
+        case IntegrationPointValue::DerivedQuantity:
+            break;
+        }
+    }
+
+private:
+    std::vector<ShapeMatrices> _shape_matrices;
+    std::vector<double>        _int_pt_values;
+};
+
+template<typename GlobalSetup>
+class TestProcess
+{
+public:
+    using GlobalMatrix = typename GlobalSetup::MatrixType;
+    using GlobalVector = typename GlobalSetup::VectorType;
+
+    using LocalAssembler = LocalAssemblerDataInterface<GlobalMatrix, GlobalVector>;
+    using GlobalAssembler = AssemblerLib::VectorMatrixAssembler<
+        GlobalMatrix, GlobalVector, LocalAssembler,
+        // The exact tag does not matter here.
+        NumLib::ODESystemTag::FirstOrderImplicitQuasilinear>;
+
+    using ExtrapolatorInterface =
+        NumLib::Extrapolator<GlobalVector, IntegrationPointValue, LocalAssembler>;
+    using ExtrapolatorImplementation =
+        NumLib::LocalLinearLeastSquaresExtrapolator<
+            GlobalVector, IntegrationPointValue, LocalAssembler>;
+
+    TestProcess(MeshLib::Mesh const& mesh)
+        : _mesh_subset_all_nodes(mesh, &mesh.getNodes())
+    {
+        std::vector<std::unique_ptr<MeshLib::MeshSubsets>> all_mesh_subsets;
+        all_mesh_subsets.emplace_back(
+                    new MeshLib::MeshSubsets{&_mesh_subset_all_nodes});
+
+        _dof_table.reset(new AssemblerLib::LocalToGlobalIndexMap(
+              std::move(all_mesh_subsets),
+              AssemblerLib::ComponentOrder::BY_COMPONENT));
+
+        // Passing _dof_table works, because this process has only one variable
+        // and the variable has exactly one component.
+        _extrapolator.reset(new ExtrapolatorImplementation(
+            { 0u, 0u, nullptr, _dof_table.get(), &mesh }));
```

Comment:
> Improve matrix spec ctor.

@endJunction, you could do that when resolving the dependency of matrix specs on the doftable.